gcloud/datastore/test_helpers.py

```diff
@@ -19,20 +19,18 @@ class Test_entity_from_protobuf(unittest2.TestCase):
 
     def setUp(self):
         from gcloud.datastore import _implicit_environ
-        self._replaced_dataset = _implicit_environ.DATASET
         self._replaced_dataset_id = _implicit_environ.DATASET_ID
-        _implicit_environ.DATASET = _implicit_environ.DATASET_ID = None
+        _implicit_environ.DATASET_ID = None
 
     def tearDown(self):
         from gcloud.datastore import _implicit_environ
-        _implicit_environ.DATASET = self._replaced_dataset
         _implicit_environ.DATASET_ID = self._replaced_dataset_id
 
     def _callFUT(self, val):
         from gcloud.datastore.helpers import entity_from_protobuf
         return entity_from_protobuf(val)
 
-    def test_wo_dataset(self):
+    def test_wo_dataset_id(self):
```

Comment:
Nice find, should I file a bug? Handle it here? You care to handle in cherry-pick of #499?